aws/resource_aws_storagegateway_smb_file_share.go

```diff
@@ -157,6 +166,14 @@ func resourceAwsStorageGatewaySmbFileShareCreate(d *schema.ResourceData, meta in
 		input.KMSKey = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("audit_destination_arn"); ok && v.(string) != "" {
```

Comment:
Nit: The zero value check is extraneous since it is using `GetOk()`. We should setup tfproviderlint for this. 😅 

```suggestion
	if v, ok := d.GetOk("audit_destination_arn"); ok {
```